### PR TITLE
Add new ATM API versions.

### PR DESCRIPTION
**Pull request by tatlicioglu**
Add new ATM API versions.

### CODE REVIEW

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -169,6 +169,9 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-03-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-05-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-02-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >trafficManagerProfiles [](start = 128, length = 22)

I don't see this resource type in 2018-02-01/Microsoft.Network.json

File: schemas/2015-01-01/deploymentTemplate.json [hunk 2]
```diff
@@ -169,6 +169,9 @@
...
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-03-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >Microsoft.Network.json [](start = 84, length = 22)

I don't see the 2018-03-01 and 2018-04-01 version of Microsoft.Network schema.

**Comment by tatlicioglu** [comment]
@TianoMS I've resolved your comments, could you please review changes again?

File: schemas/2018-04-01/Microsoft.Network.json [hunk 3]
```diff
@@ -0,0 +1,547 @@
+{
+    "id": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Network.json#",
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "title": "Microsoft.Network",
+    "description": "Microsoft Network Resource Types",
+    "resourceDefinitions": {
+      "trafficmanagerprofiles": {
```
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Network.json]: >trafficmanagerprofiles [](start = 7, length = 22)

please add tests in /tests folder for the new resource types.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 4]
```diff
@@ -169,6 +169,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-11-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-03-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
                   { "$ref": "https://schema.management.azure.com/schemas/2017-05-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >trafficManagerProfiles [](start = 128, length = 22)

this needs to be added to 2014-04-01-preview/deploymentTemplate.json as well.

File: schemas/2018-04-01/Microsoft.Network.json (hunk 3 above)
Comment (TianoMS) [inline schemas/2018-04-01/Microsoft.Network.json]: >trafficmanagerprofiles [](start = 7, length = 22)

Please use consistent resource type names.. In the root template it's "trafficManagerProfiles"

File: tests/2018-04-01/Microsoft.Network.tests.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+    "tests": [
+      {
+        "name": "Traffic Manager profile with one endpoint of each type and optional properties",
+        "definition": "https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles",
```
Comment (TianoMS) [inline tests/2018-04-01/Microsoft.Network.tests.json]: >trafficManagerProfiles [](start = 122, length = 22)

the test case is failing..

**Comment by tatlicioglu** [comment]
Thanks @TianoMS 

I've fixed the casing and I see locally all tests are passing fine:
2342 tests passed
0 tests failed

**Comment by Tiano2017** [comment]
got an message: "Problems loading reference 'https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Network.json#/resourceDefinitions/trafficManagerProfiles': Unable to parse content from 'https://schema.management.azure.com/schemas/2018-04-01/Microsoft.Network.json': Parse error at offset 276."

**Comment by Tiano2017** [comment]
we are enabling the new template test we have. To run it: go to "/tools", run "npm test".

**Comment by Tiano2017** [comment]
I made a mistake. I think this PR is clear now.

**Pull request merged**